File: backend/app/routers/health.py
from fastapi import APIRouter
from datetime import datetime
from app import get_version

router = APIRouter()

@router.get("/health")
async def health_check():
    return {
        "status": "healthy",
        "timestamp": datetime.utcnow().isoformat(),
        "service": "navidrome-toolbox-api",
        "version": get_version(),
    }

File: frontend/lib/api/health.ts
import { HealthResponse } from '@/types/api';

export async function checkHealth(): Promise<HealthResponse> {
  try {
    const res = await fetch('/api/health', { cache: 'no-store' });
    if (!res.ok) throw new Error();
    return await res.json();
  } catch {
    return { 
      status: 'unhealthy', 
      timestamp: new Date().toISOString(), 
      service: 'navidrome-toolbox-api', 
      version: 'unknown' 
    };
  }
}
